package/piwebcam/multi-gadget.sh

```diff
@@ -64,17 +66,26 @@ EOF
 config_usb_webcam () {
   mkdir -p functions/uvc.usb0/control/header/h
 
-  config_frame mjpeg m  640  360
-  config_frame mjpeg m  640  480
-  config_frame mjpeg m  800  600
-  config_frame mjpeg m 1024  768
-  config_frame mjpeg m 1280  720
-  config_frame mjpeg m 1280  960
-  config_frame mjpeg m 1440 1080
-  config_frame mjpeg m 1536  864
-  config_frame mjpeg m 1600  900
-  config_frame mjpeg m 1600 1200
-  config_frame mjpeg m 1920 1080
+  if [ -f "$RESOLUTION_FILE" ] ; then
+    sed "s/ //g" "$RESOLUTION_FILE" | grep "\S" | grep -vE "^\#" | while read -r line
```

Comment:
I am pondering about the right format of that file. One might want to use `raw` instead of `mjpeg` or even `mp4` if uvc-gadget gets support for that. In that case it would be better to add the format to that file as well. Also, it's quite error prone to parse it like that.

How about:

```
mjpeg 640 360
mjpeg 640 480
...
```
That way you could parse it with awk and drop that line if it does not contain out of 3 words. (Bonus point if you check for valid formats and if the latter two arguments are integers)

This would then for example allow to easily use `raw` etc. As for the configfs name you'd then just use the first letter of the format.

Further opinions welcome!

Reply:
I will have a look at this. I would then rename the input file from `resolutions.txt` to `output_formats.conf`.

Reply:
Let's keep the "txt" extensions as all the other files do use that too. Looking forward to your ideas.